tests/upgrades/test_containers_removal.py

```diff
@@ -0,0 +1,209 @@
+            self.assertEqual(external_container.repository_name, repo_name)
+
+            running_containers = docker_host.run('docker ps')
+            self.assertEqual(running_containers.return_code, 0)
+
+            self.assertTrue(any(dockerhub_container.name in line
+                                for line in running_containers.stdout))
+            self.assertTrue(any(external_container.name in line
+                                for line in running_containers.stdout))
+
+            scenario_dict = {self.__class__.__name__: {
+                'docker_host': docker_host.hostname,
+                'dockerhub_container': dockerhub_container.name,
+                'external_container': external_container.name,
+            }}
+            create_dict(scenario_dict)
+        except Exception as exp:
+            self._vm_cleanup(hostname=docker_host.hostname)
+            raise Exception(exp)
+
+    @post_upgrade(depend_on=test_pre_scenario_containers_support_removal)
+    def test_post_scenario_containers_support_removal(self):
+        """Post-upgrade scenario test to verify containers created and run
+        before upgrade are still running after upgrade.
+
+        :id: postupgrade-f6de07ae-14c7-4452-9cb1-cafe2aa648ae
+
+        :steps:
+
+            1. Verify that upgrade procedure removed foreman_docker support
+            2. Verify container from dockerhub is still running
+            3. Verify container from external registry is still running
+
+        :expectedresults:
+
+            1. Upgrade procedure removed support for foreman_docker
+            2. Container from dockerhub is still running
+            3. Container from external registry is still running
+
+        """
+        entity_data = get_entity_data(self.__class__.__name__)
+        docker_host_hostname = entity_data.get('docker_host')
+        dockerhub_container = entity_data.get('dockerhub_container')
+        external_container = entity_data.get('external_container')
+
+        extract_log_command = "sed -n '/{delimiter}/,/{delimiter}/p' {path}".format(
+            delimiter="RemoveForemanDockerSupport",
+            path="/var/log/foreman-installer/satellite.log"
+        )
+        docker_log = ssh.command(extract_log_command, output_format='plain')
+        self.assertEqual(docker_log.return_code, 0)
+
+        docker_log = docker_log.stdout.split("\n")
+
+        self.assertTrue(any("RemoveForemanDockerSupport: migrated" in line
+                            for line in docker_log))
+        self.assertTrue(any("remove_column(:containers, :capsule_id)" in line
+                            for line in docker_log))
+
+        removed_tables = ('docker_images', 'docker_tags',
+                          'docker_container_wizard_states_images',
+                          'containers')
+
+        for table in removed_tables:
+            self.assertTrue(any("table_exists?(:{})".format(table)
+                                in line for line in docker_log))
+
```

Comment:
This one is tricky. Removal of support for docker operations in Robottelo (CLI), Nailgun and Airgun is on my list of things to do. Once these commits are merged, our tooling will be unable to call these endpoints and verify that they are indeed removed on Satellite side.

My idea was to create separate test cases that would ensure endpoints are removed. This will be easy for CLI and doable for API; not yet sure how I can approach web UI. We can decorate these test cases with `@upgrade` to ensure they are run after upgrade, too.

Another approach would be to leave support for docker operations in our tooling for now, and remove it sometime after 6.6 GA. This way I can call API endpoint and expect exception.

What do you think?